src/v/datalake/coordinator/state_update.cc

```diff
@@ -45,12 +49,34 @@ add_files_update::can_apply(const topics_state& state) {
     if (entries.empty()) {
         return stm_update_error{"No entries requested"};
     }
-    auto prt_state_opt = state.partition_state(tp);
-    if (!prt_state_opt.has_value()) {
-        // No entries at all, this partition hasn't ever added any files.
+    auto topic_it = state.topic_to_state.find(tp.topic);
+    if (topic_it == state.topic_to_state.end()) {
+        return stm_update_error{fmt::format(
+          "topic {} rev {} not yet registered", tp.topic, topic_revision)};
+    }
+    auto& cur_topic = topic_it->second;
+    if (topic_revision < cur_topic.revision) {
+        // Files will be added to the orphan_files collection.
```

Comment:
It seems wrong for the coordinator to shoulder orphan file cleanup on stale revision, instead of rejecting the STM update and rejecting translator RPCs, and making the translators clean up after themselves (which IIUC they already do)

Reply:
> which IIUC they already do

Currently no, they don't clean the file from cloud storage on coordinator errors.